storage/innobase/srv/srv0start.cc

```diff
@@ -580,11 +584,82 @@ static void create_log_files_rename(
   strcpy(logfile0, logfilename);
 
   fil_open_log_and_system_tablespace_files();
+  fil_space_t::s_redo_space = fil_space_get(dict_sys_t::s_log_space_first_id);
 
   /* For cloned database it is normal to resize redo logs. */
   ib::info(ER_IB_MSG_1068, ulonglong{lsn});
 }
 
+/* create the log file name
+@param[in/out] logfilename  buffer for log file name
+retrun the dir name lenght */
+static size_t create_log_file_name(char *logfilename, size_t logfilename_size) {
+  size_t dirnamelen;
+  dirnamelen = strlen(srv_log_group_home_dir);
+  ut_a(dirnamelen < logfilename_size - 10 - sizeof "ib_logfile");
+  memcpy(logfilename, srv_log_group_home_dir, dirnamelen);
+
+  /* Add a path separator if needed. */
+  if (dirnamelen && logfilename[dirnamelen - 1] != OS_PATH_SEPARATOR) {
+    logfilename[dirnamelen++] = OS_PATH_SEPARATOR;
+  }
+  return dirnamelen;
+}
+
+/* recreate the redo logs
+@param[in,out]  logfilename     buffer for log file name
+@param[in]      dirnamelen      length of the directory path
+@param[out]     logfile0        name of the first log file
+@param[out]     checkpoint_lsn  lsn of the first created checkpoint
+@param[in]      flushed_lsn     current flushed lsn
+@param[in]      num_of_files    number of redo log files to be created
+@return DB_SUCCESS or error code */
+static dberr_t recreate_redo_logs(char *logfilename, size_t dirnamelen,
+                                  char *&logfile0, lsn_t new_checkpoint_lsn,
+                                  lsn_t flushed_lsn,
+                                  uint32_t number_of_log_files) {
+  ut_ad(fil_space_t::s_redo_space != nullptr);
+  /* Close and free the redo log files, so that
+  we can replace them. */
+  fil_close_log_files(true);
+
+  RECOVERY_CRASH(5);
+
+  log_sys_close();
+
+  /* Finish clone file recovery before creating new log files. We
+      roll forward to remove any intermediate files here. */
+  clone_files_recovery(true);
+
+  auto err =
+      create_log_files(logfilename, dirnamelen, flushed_lsn,
+                       number_of_log_files, logfile0, new_checkpoint_lsn);
+
+  if (err != DB_SUCCESS) {
+    return err;
+  }
+
+  /* create_log_files() can increase system lsn that is
+      why FIL_PAGE_FILE_FLUSH_LSN have to be updated */
```

Comment:
create_log_files() can increase LSN, update the FIL_PAGE_FILE_FLUSH_LSN to the last LSN

Reply:
fixed